src/wireviz/wireviz.py

```diff
@@ -44,10 +45,11 @@ def parse(yaml_input: str, file_out: (str, Path) = None, return_types: (None, st
             if len(yaml_data[sec]) > 0:
                 if ty == dict:
                     for key, attribs in yaml_data[sec].items():
-                        # The Image dataclass might need to open an image file with a relative path.
                         image = attribs.get('image')
                         if isinstance(image, dict):
-                            image['gv_dir'] = Path(file_out if file_out else '').parent # Inject context
+                            image_path = image['src']
+                            if image_path and not Path(image_path).is_absolute():  # resolve relative image path
+                                image['src'] = (Path(base_path) / image_path).resolve()
```

Comment:
Thanks @kvid, I've included the suggestion from your [comment](https://github.com/formatc1702/WireViz/pull/189#discussion_r532944282).

I'm a little confused why `image` is a *reference* to `attribs` and therefore assigning `image[src] = ...` directly affects `yaml_data`'s contents; where as `image_path` just gets passed the *value* of the dict key, and `image_path = ...` does not affect `yaml_data`.

Even if using `image_path = image.get('src')` in line 50 (analogously to `image = attribs.get('image')` in line 48), you cannot do the final assignment with `image_path = (Path(base_path) / image_path).resolve()`.

Maybe I'm missing something... can you explain why `image` is a reference and `image_path` is not? Thanks :)

Reply:
> Thanks @kvid, I've included the suggestion from your [comment](https://github.com/formatc1702/WireViz/pull/189#discussion_r532944282).

Thank you for accepting my suggestion.

> 
> I'm a little confused why `image` is a _reference_ to `attribs` and therefore assigning `image[src] = ...` directly affects `yaml_data`'s contents; where as `image_path` just gets passed the _value_ of the dict key, and `image_path = ...` does not affect `yaml_data`.
> 
> Even if using `image_path = image.get('src')` in line 50 (analogously to `image = attribs.get('image')` in line 48), you cannot do the final assignment with `image_path = (Path(base_path) / image_path).resolve()`.
> 
> Maybe I'm missing something... can you explain why `image` is a reference and `image_path` is not? Thanks :)

Both `image` and `image_path` are references to objects that also are reachable through `yaml_data`.
- The `image` variable is a reference to a `dict` that is mutable, and this `dict` can be changed through the reference, e.g. `image['src'] = 'new value'`, and the changed object is also are reachable through `yaml_data` (the reference is unchanged).
- However, The `image_path` variable is a reference to a `str` that is immutable, and immutable objects cannot be changed. Assigning a different value to the `image_path` variable will make it a reference to a different object, but that does not affect the original referenced immutable object (that is still reachable through `yaml_data`).

I hope my description above is somewhat understandable, and don't create even more confusion. Others have described this differences in more detail:
- https://www.freecodecamp.org/news/mutable-vs-immutable-objects-python/
- https://medium.com/@meghamohan/mutable-and-immutable-side-of-python-c2145cf72747

Reply:
Thanks for the explanation and links! Very helpful. Will mark as resolved.